src/index.js

```diff
@@ -1,390 +1,586 @@
-export function h(name, attributes) {
-  var rest = []
-  var children = []
-  var length = arguments.length
+var DEFAULT = 0
+var RECYCLED_NODE = 1
+var TEXT_NODE = 2
 
-  while (length-- > 2) rest.push(arguments[length])
+var XLINK_NS = "http://www.w3.org/1999/xlink"
+var SVG_NS = "http://www.w3.org/2000/svg"
 
-  while (rest.length) {
-    var node = rest.pop()
-    if (node && node.pop) {
-      for (length = node.length; length--; ) {
-        rest.push(node[length])
+var EMPTY_OBJECT = {}
+var EMPTY_ARRAY = []
+
+var map = EMPTY_ARRAY.map
+var isArray = Array.isArray
+
+var merge = function(a, b) {
+  var target = {}
+
+  for (var i in a) target[i] = a[i]
+  for (var i in b) target[i] = b[i]
+
+  return target
+}
+
+var resolved = typeof Promise === "function" && Promise.resolve()
+
+var defer = !resolved
+  ? setTimeout
+  : function(cb) {
+      return resolved.then(cb)
+    }
+
+var updateProperty = function(
+  element,
+  name,
+  lastValue,
+  nextValue,
+  eventProxy,
+  isSvg
+) {
+  if (name === "key") {
```

Comment:
Could also be 
```js
if(name === "style") {
  ...
} else if (name !== "key") {
  ...
}
```
To avoid an empty if

Reply:
I'm a fan of the "exit early" pattern
```
if (name === "key") return

// Do other ifs
```

Reply:
I wonder if this is just personal taste or there's something to it. My brain finds it a lot easier to think if something is true, instead of if not something do something.